src/puppetlabs/trapperkeeper/services/webserver/jetty9_core.clj

```diff
@@ -270,25 +273,50 @@
   (connector* server config nil))
 
 (schema/defn ^:always-validate
-  queue-thread-pool :- QueuedThreadPool
-  [max-threads :- schema/Int
-   queue-max-size :- schema/Int]
-  (let [queue-min-size (min config/default-queue-min-threads queue-max-size)]
-    (QueuedThreadPool. max-threads
-                       queue-min-size
-                       config/default-queue-idle-timeout
-                       (BlockingArrayQueue.
-                         queue-min-size
-                         queue-min-size
-                         queue-max-size))))
+  queue-thread-pool :- (schema/maybe QueuedThreadPool)
+  [max-threads :- (schema/maybe schema/Int)
+   queue-max-size :- (schema/maybe schema/Int)]
+  (if (or max-threads queue-max-size)
+    (let [thread-pool (if max-threads
+                        (QueuedThreadPool. max-threads)
+                        (QueuedThreadPool.))]
+      (if queue-max-size
+        ;; The code below is definitely not ideal, but there isn't a way to set
+        ;; the maximum capacity of the QueuedThreadPool's BlockingArrayQueue
+        ;; after construction.  We're trying to avoid hard-coding our own
+        ;; defaults for other settings that we want Jetty to control, e.g., the
+        ;; initial capacity of the queue and minimum number of threads.  By
+        ;; reconstructing the QueuedThreadPool here, we can use Jetty's defaults
+        ;; for settings unrelated to `queue-max-size`.  QueuedThreadPool and
+        ;; BlockingArrayQueue construction isn't too expensive.  It mostly
+        ;; involves some initial memory allocations.  The more expensive work -
+        ;; where threads are actually started and the queue expands to fulfill
+        ;; new requests - would only happen after the QueuedThreadPool were
+        ;; started.  That won't happen for the `thread-pool` instance from
+        ;; above, which just gets thrown away as this function falls out of
+        ;; scope without ever having been started.  Also, this function would
+        ;; likely only be called once per server startup where a
+        ;; `queue-max-size` were configured.
+        (let [min-threads    (.getMinThreads thread-pool)
+              queue-min-size (min queue-max-size min-threads)]
+          (QueuedThreadPool. (.getMaxThreads thread-pool)
+                             min-threads
+                             (.getIdleTimeout thread-pool)
+                             (BlockingArrayQueue.
+                               queue-min-size
+                               queue-min-size
+                               queue-max-size)))
```

Comment:
> However, in the second link you provided above, notice that the growth rate is not specified - so I'm still not sure why we are setting that, and why we're using queue-min-size as the value.

Not sure I follow.  Jetty's `QueuedThreadPool`, by default, is calling the form of the `BlockingArrayQueue` constructor that takes in a "capacity" and a "growBy".  See https://github.com/eclipse/jetty.project/blob/jetty-9.2.10.v20150310/jetty-util/src/main/java/org/eclipse/jetty/util/BlockingArrayQueue.java#L113.  It sets them both to the same value, which derives from the minimum number of threads in the `QueuedThreadPool`.

I could define a `queue-capacity` and `queue-grow-by` and set them both to the result of `(min queue-max-size min-threads)` if that would help for clarity, although that wouldn't produce a functional difference.